src/Optimizer/Transformer/DetermineHeroImages.php

```diff
@@ -181,14 +189,27 @@ private function get_initial_content_image( Document $document ) {
 			return null;
 		}
 
-		$query = $document->xpath->query(
-			self::INITIAL_CONTENT_IMAGE_XPATH_QUERY,
-			$entry_content
-		);
-
+		$query = $document->xpath->query( self::INITIAL_CONTENT_IMAGE_XPATH_QUERY, $entry_content );
 		$image = $query->item( 0 );
-		if ( $image instanceof Element && ! ( new ImageDimensions( $image ) )->isTiny() ) {
-			return $image;
+
+		if ( ! $image instanceof Element ) {
+			return null;
+		}
+
+		if ( Extension::IMG === $image->tagName ) {
+			return ! ( new ImageDimensions( $image ) )->isTiny() ? $image : null;
+		}
+
+		$style = $image->getAttribute( Attribute::STYLE );
+		$style = ! empty( $style ) ? $style : $image->getAttribute( 'data-amp-original-style' );
+
+		if ( ! empty( $style ) ) {
+			preg_match( self::CSS_BACKGROUND_IMAGE_URL_REGEX_PATTERN, $style, $matches );
+
+			if ( ! empty( $matches['url'] ) ) {
+				$image->setAttribute( Attribute::MEDIA, $matches['url'] );
+				return $image;
+			}
```

Comment:
This doesn't seem to work. In testing I see the following `link` being added to the `head`:

```HTML
<link rel="preload" href="https://wordpressdev.lndo.site/content/uploads/2021/11/PXL_20211126_001920318.NIGHT_-scaled.jpg" as="image" data-hero media="https://wordpressdev.lndo.site/content/uploads/2021/11/PXL_20211126_001920318.NIGHT_-scaled.jpg">
```

Note the `media` attribute is set to something invalid. This causes the image to never even load.

Reply:
I have updated the value of the media attribute to the `screen`.